spec/models/user_spec.rb

```diff
@@ -56,59 +55,55 @@
     end
   end
 
-  describe "email validation" do
+  describe "email" do
     # other email validations are done by devise and not tested here.
     # Devise ignores the max length requirement of RFC 3696, errata ID 1690
     # so there is an explicit length validation we added and are testing
-    describe "basic validation" do
-      it "fails if the email is > 254 characters" do
-        suffix = ".com"
-        @user.email = "a@long"
-        @user.email += "g" * (255 - @user.email.length - suffix.length)
-        @user.email += suffix
-        expect(@user).not_to be_valid
-      end
+    it "must be <= 254 characters" do
+      suffix = ".com"
+      @user.email = "a@long"
+      @user.email += "g" * (255 - @user.email.length - suffix.length)
+      @user.email += suffix
+      expect(@user).not_to be_valid
+    end
 
-      it "fails if an email is not unique" do
-        user2 = User.new(name: @user.name, email: @user.email)
-        @user.save
-        expect(user2).not_to be_valid
-      end
+    it "must be unique" do
+      user2 = User.new(name: @user.name, email: @user.email)
+      @user.save
+      expect(user2).not_to be_valid
+    end
 
-      it "email uniqueness is case-insensitive" do
-        user2 = User.new(name: @user.name, email: @user.email.upcase)
-        @user.save
-        expect(user2).not_to be_valid
-      end
+    it "uniqueness is case-insensitive" do
+      user2 = User.new(name: @user.name, email: @user.email.upcase)
+      @user.save
+      expect(user2).not_to be_valid
     end
 
-    describe "format validation" do
-      it "accepts email addresses with proper format" do
-        valid_addresses = %w[ user@example.com USER@foo.COM A_US-ER@foo.bar.org first.last@foo.jp alice+bob@baz.cn ]
-        valid_addresses.each do |addr|
-          @user.email = addr
-          expect(@user).to be_valid
-        end
+    it "accepts addresses with proper format" do
```

Comment:
Not orginally, but I switched over. I kept this test since it was already written.